packages/amplify-category-function/src/provider-utils/awscloudformation/utils/layerArnConverter.ts

```diff
@@ -16,13 +19,18 @@ export const convertLambdaLayerMetaToLayerCFNArray = (input: LambdaLayer[], env:
   });
 };
 
-const convertProjectLayer = (layer: ProjectLayer, env?: string) => {
-  if (env) {
+export const convertProjectLayer = (layer: ProjectLayer, env?: string) => {
+  if (!layer.isLatestVersionSelected) {
+    if (env) {
```

Comment:
I am not sure if we need to handle no environment like this as I am not convinced that when you get `envName` that would not fail, please check.

Reply:
The check is for is for legacy non-multienv layers. We should add a comment for clarity

Reply:
yeah , we are calling this function in update layer also I guess to support non-multienv layers